src/blockchain/compact_filters/nakamoto.rs

```diff
@@ -0,0 +1,687 @@
+
+    database.commit_batch(updates)?;
+
+    Ok(())
+}
+
+/// Mark as transaction as unconfirmed in the database.
+/// This is called in block reorg cases, where previously confirmed transactions becomes unconfirmed.
+fn unconfirm_tx(
+    database: &mut impl BatchDatabase,
+    txid: &Txid,
+) -> Result<TransactionDetails, CbfError> {
+    let mut tx_details = database
+        .get_tx(&txid, false)?
+        .expect("We must have the transaction at this stage");
+    tx_details.confirmation_time = None;
+    database.set_tx(&tx_details)?;
+    debug!("Existing transaction marked unconfirmed : {}", txid);
+    Ok(tx_details)
+}
+
+/// Delete a Transaction from the database, updating related UTXO records.
+/// This is called in a a RBF case.
+fn delete_tx(database: &mut impl BatchDatabase, txid: &Txid) -> Result<(), CbfError> {
+    let mut updates = database.begin_batch();
+
+    if let Some(db_tx) = database.get_raw_tx(txid)? {
+        let txid = db_tx.txid();
+        // Mark input utxos as unspent
+        for input in db_tx.input {
+            let prev_output = input.previous_output;
+            if let Some(mut existing_utxo) = database.get_utxo(&prev_output)? {
+                debug!("Setting UTXO : {} as `unspent`", existing_utxo.outpoint);
+                existing_utxo.is_spent = false;
+                updates.set_utxo(&existing_utxo)?;
+            }
+        }
+
+        // Delete output utxos
+        for (i, _) in db_tx.output.iter().enumerate() {
+            let outpoint = OutPoint::new(txid, i as u32);
+            if let Some(deleted_utxo) = updates.del_utxo(&outpoint)? {
+                debug!("Deleting Existing UTXO {}", deleted_utxo.outpoint);
+            }
+        }
+
+        // Delete the transaction data
+        let _ = database
+            .del_tx(&txid, true)?
+            .expect("We must have the transaction at this stage")
+            .txid;
+        debug!("Deleting existing transaction : txid: {}", txid);
+    }
+
+    database.commit_batch(updates)?;
+
+    Ok(())
+}
+
+/// Structure containing the nakamoto client handle and event receiver.
+pub struct CbfNode {
+    /// A Receiver of all client related events
+    receiver: Receiver<Event>,
+    /// The handler for the client
+    client_handle: ClientHandle<Waker>,
+    /// Internal query timeout. Set by default at 60 secs.
+    timeout: Duration,
+    /// A store of blockheight and fee estimation
+    fee_data: Cell<HashMap<u32, FeeEstimate>>,
+    /// A local cache of unconfirmed broadcasted transactions.
+    /// Used to sync wallet db with known unconfirmed txs. Cleared at each `wallet.sync()` call.
+    broadcasted_txs: Cell<Vec<Transaction>>,
+    /// Last Sync height
+    last_sync_height: Cell<u32>,
+    // A Hack to handle reorg sync cases.
+    #[cfg(test)]
+    break_sync_at: Cell<Option<u32>>,
+}
+
+/// Nakamoto CBF Client Configuration
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+pub struct CBFConfig {
```

Comment:
small nit: I think to keep in line with the other two `ElectrumBlockchainConfig` and `EsploraBlockchainConfig` you'd want to name this one `CbfBlockchainConfig`, or maybe `NakamotoBlockchainConfig` (just a suggestion).

Reply:
Done..